pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/http/AsyncHttpConnector.java

```diff
@@ -76,63 +86,102 @@ public boolean keepAlive(Request ahcRequest, HttpRequest request, HttpResponse r
             }
         });
 
-        if (conf != null && StringUtils.isNotBlank(conf.getServiceUrl())
-                && conf.getServiceUrl().startsWith("https://")) {
-
-            SslContext sslCtx = null;
-
-            // Set client key and certificate if available
-            AuthenticationDataProvider authData = conf.getAuthentication().getAuthData();
-            if (authData.hasDataForTls()) {
-                sslCtx = SecurityUtility.createNettySslContextForClient(
-                        conf.isTlsAllowInsecureConnection() || !conf.isTlsHostnameVerificationEnable(),
-                        conf.getTlsTrustCertsFilePath(),
-                        authData.getTlsCertificates(), authData.getTlsPrivateKey());
-            } else {
-                sslCtx = SecurityUtility.createNettySslContextForClient(
-                        conf.isTlsAllowInsecureConnection() || !conf.isTlsHostnameVerificationEnable(),
-                        conf.getTlsTrustCertsFilePath());
-            }
+        serviceNameResolver = new PulsarServiceNameResolver();
+        if (conf != null && StringUtils.isNotBlank(conf.getServiceUrl())) {
+            serviceNameResolver.updateServiceUrl(conf.getServiceUrl());
+            if (conf.getServiceUrl().startsWith("https://")) {
+
+                SslContext sslCtx = null;
+
+                // Set client key and certificate if available
+                AuthenticationDataProvider authData = conf.getAuthentication().getAuthData();
+                if (authData.hasDataForTls()) {
+                    sslCtx = SecurityUtility.createNettySslContextForClient(conf.isTlsAllowInsecureConnection() || !conf.isTlsHostnameVerificationEnable(),
+                                                                            conf.getTlsTrustCertsFilePath(), authData.getTlsCertificates(), authData.getTlsPrivateKey());
+                } else {
+                    sslCtx = SecurityUtility.createNettySslContextForClient(conf.isTlsAllowInsecureConnection() || !conf.isTlsHostnameVerificationEnable(),
+                                                                            conf.getTlsTrustCertsFilePath());
+                }
 
-            confBuilder.setSslContext(sslCtx);
+                confBuilder.setSslContext(sslCtx);
+            }
         }
         httpClient = new DefaultAsyncHttpClient(confBuilder.build());
     }
 
-    @Override
     public ClientResponse apply(ClientRequest jerseyRequest) {
-        CompletableFuture<ClientResponse> future = new CompletableFuture<>();
 
-        try {
-            Future<?> resultFuture = apply(jerseyRequest, new AsyncConnectorCallback() {
-                @Override
-                public void response(ClientResponse response) {
-                    future.complete(response);
+        List<InetSocketAddress> addresses = serviceNameResolver.getAddressList();
```

Comment:
I would suggest shuffling the list of addresses so that the request can be sent to different brokers.

Reply:
I get all address and try it one by one